stock_free_quantity/models/product_template.py

```diff
@@ -0,0 +1,35 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _compute_quantities_dict(self):
+        prod_available = super()._compute_quantities_dict()
+        for template in self:
+            free_qty = 0
+            for p in template.with_context(active_test=False).product_variant_ids:
+                free_qty += p.free_qty
+            prod_available[template.id]["free_qty"] = free_qty
+        return prod_available
+
+    def _compute_quantities(self):
+        super()._compute_quantities()
+        res = self._compute_quantities_dict()
+        for template in self:
+            template.free_qty = res[template.id]["free_qty"]
+
+    def _search_free_qty(self, operator, value):
+        domain = [("free_qty", operator, value)]
+        product_variant_ids = self.env["product.product"].search(domain)
+        return [("product_variant_ids", "in", product_variant_ids.ids)]
```

Comment:
You can avoid a search returning directly the domain [("product_variant_ids.free_qty", operator, value)].

But I think this assertion is False as it does not aggregate product variants free qty for the template but search on each one.

Reply:
Returning  `[("product_variant_ids.free_qty", operator, value)]`, the search will then be done after anyway right ? (in variant search method)
Anyway, I confess I just copied/pasted the method defined in stock module for the other template quantity fields.
I think we have to keep the same logic (not aggregating the variant free qty) for consistency, what do you think ?

Reply:
That depends on what this free_qty means on product template.

If you want to search templates that have variants' free qty ... values, then it's ok. But if you want to search templates that have free qty according to aggregated values, then it should be changed.

In the case of first solution, change _search method as I mentionned. It will avoid a search (your solution do a first search then  a second one after having returned the domain).

Reply:
Ok, I changed the domain, and went for the first solution, as it is the Odoo native solution for all quantity fields on templates.